account_banking_payment_transfer/model/account_payment.py

```diff
@@ -241,38 +246,28 @@ def action_sent(self):
             # key = unique identifier (date or True or line.id)
             # value = [pay_line1, pay_line2, ...]
             trfmoves = {}
-            if self.mode.transfer_move_option == 'line':
-                for line in self.line_ids:
-                    trfmoves[line.id] = [line]
-            else:
-                if self.date_prefered in ('now', 'fixed'):
-                    trfmoves[True] = []
-                    for line in self.line_ids:
-                        trfmoves[True].append(line)
-                else:  # date_prefered == due
-                    for line in self.line_ids:
-                        if line.date in trfmoves:
-                            trfmoves[line.date].append(line)
-                        else:
-                            trfmoves[line.date] = [line]
+            for bline in self.bank_line_ids:
+                hashcode = bline.move_line_transfer_account_hashcode()
+                if hashcode in trfmoves:
+                    trfmoves[hashcode].append(bline)
+                else:
+                    trfmoves[hashcode] = [bline]
 
-            for identifier, lines in trfmoves.iteritems():
+            for hashcode, blines in trfmoves.iteritems():
                 mvals = self._prepare_transfer_move()
                 move = am_obj.create(mvals)
                 total_amount = 0
-                for line in lines:
-                    total_amount += line.amount
-                    self._create_move_line_partner_account(line, move, labels)
+                for bline in blines:
+                    total_amount += bline.amount_currency
+                    self._create_move_line_partner_account(bline, move, labels)
                 # create the payment/debit move line on the transfer account
                 trf_ml_vals = self._prepare_move_line_transfer_account(
-                    total_amount, move, lines, labels)
+                    total_amount, move, blines, labels)
                 aml_obj.create(trf_ml_vals)
-                self._reconcile_payment_lines(lines)
+                self._reconcile_payment_lines(blines)
 
-                # consider entry_posted on account_journal
-                if move.journal_id.entry_posted:
```

Comment:
The help pop-up of entry_posted says: 'Check this box to automatically post entries of this journal. Note that legally, some entries may be automatically posted when the source document is validated (Invoices), whatever the status of this field.'

When you validate an invoice or a bank statement in Odoo, the entry is automatically posted, even when entry_posted=False. I feel that it should be the same for the transfer move. It would be difficult to explain to accountants that this option is ignored for invoices and bank statements, and it is taken into account for the transfer move of a payment.

For me, the entry_posted option is taken into account for moves that are manually created by the user.